File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> vecs;
        if(!root){
            return vecs;
        }
        queue<TreeNode*> q;
        q.push(root);
        while(!q.empty()) {
            vector<int> vec;
            int size = q.size();
            for(int i = size; i>0; i--){
                TreeNode* node = q.front();
                q.pop();
                if(node){
                vec.push_back(node->val);
                q.push(node->left);
                q.push(node->right);
                }
            }

            if(!vec.empty()){
                vecs.push_back(vec);
            }

        }
        return vecs;

    }
};

File: Data Structures & Algorithms/valid-binary-search-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        def dfs(node, low, high):
            if not node:
                return True
            if node.val<=low or node.val>=high:
                return False
            left_ok = dfs(node.left, low, node.val)
            right_ok = dfs(node.right, node.val, high)
            return left_ok and right_ok
        
        return dfs(root, float("-inf"), float("inf"))
    
        